File: src/application/contracts/RecipeRepository.ts
import { Recipe } from '@domain/entities/Recipe.js'
import { Id } from '@domain/valueobjects/Id.js'

export interface RecipeRepository {
  save(recipe: Recipe): Promise<void>
  findById(id: Id): Promise<Recipe | null>
  findByName(name: string): Promise<Recipe | null>
  findAll(): Promise<Recipe[]>
  update(recipe: Recipe): Promise<void>
  delete(id: Id): Promise<void>
  existsById(id: Id): Promise<boolean>
  existsByName(name: string): Promise<boolean>
}

File: api/src/main/java/br/unifor/costify/recipe/application/contracts/RecipeRepository.java
package br.unifor.costify.recipe.application.contracts;

import br.unifor.costify.recipe.domain.entity.Recipe;
import br.unifor.costify.shared.domain.valueobject.Id;
import java.util.List;
import java.util.Optional;

public interface RecipeRepository {

  Recipe save(Recipe recipe);

  Optional<Recipe> findById(Id id);

  List<Recipe> findAll();

  boolean existsByName(String name);

  void deleteById(Id id);

  List<Recipe> findByIngredientId(Id ingredientId);
}
